controllers/awsmachine_controller_test.go

```diff
@@ -1004,6 +1021,367 @@ func TestAWSMachineReconciler(t *testing.T) {
+					InstanceIds: aws.StringSlice([]string{"id-1"}),
+				}),
+			).
+				Return(nil, errors.New("Failed to delete instance"))
+		}
+		expect(ec2Mock.EXPECT(), elbMock.EXPECT())
+
+		ns, err := testEnv.CreateNamespace(ctx, fmt.Sprintf("integ-test-%s", util.RandomString(5)))
+		g.Expect(err).To(BeNil())
+
+		setup(t, g)
+		awsMachine := getAWSMachine()
+		awsMachine.Namespace = ns.Name
+		createAWSMachine(g, awsMachine)
+
+		defer teardown(g)
+		defer t.Cleanup(func() {
+			g.Expect(testEnv.Cleanup(ctx, awsMachine, ns)).To(Succeed())
+		})
+
+		cs, err := getClusterScope(infrav1.AWSCluster{ObjectMeta: metav1.ObjectMeta{Name: "test"}})
+		g.Expect(err).To(BeNil())
+		cs.Cluster = &clusterv1.Cluster{ObjectMeta: metav1.ObjectMeta{Name: "test-cluster"}}
+		ms, err := getMachineScope(cs, awsMachine)
+		g.Expect(err).To(BeNil())
+
+		ms.AWSMachine.Status.InstanceState = &infrav1.InstanceStateRunning
+		ms.Machine.Labels = map[string]string{clusterv1.MachineControlPlaneLabelName: ""}
+		ms.AWSMachine.Spec.ProviderID = aws.String("aws:////myMachine")
+
+		ec2Svc := ec2Service.NewService(cs)
+		ec2Svc.EC2Client = ec2Mock
+		reconciler.ec2ServiceFactory = func(scope scope.EC2Scope) services.EC2MachineInterface {
+			return ec2Svc
+		}
+
+		elbSvc := elbService.NewService(cs)
+		elbSvc.EC2Client = ec2Mock
+		elbSvc.ELBClient = elbMock
+		reconciler.elbServiceFactory = func(scope scope.ELBScope) services.ELBInterface {
+			return elbSvc
+		}
+
+		_, err = reconciler.reconcileDelete(ms, cs, cs, cs)
+		g.Expect(err).Should(HaveOccurred())
+		expectConditions(g, ms.AWSMachine, []conditionAssertion{{infrav1.InstanceReadyCondition, corev1.ConditionFalse, clusterv1.ConditionSeverityWarning, "DeletingFailed"},
+			{infrav1.ELBAttachedCondition, corev1.ConditionFalse, clusterv1.ConditionSeverityInfo, clusterv1.DeletedReason}})
+		g.Expect(ms.AWSMachine.Finalizers).ShouldNot(ContainElement(infrav1.MachineFinalizer))
+	})
+}
+
+func getMachineScope(cs *scope.ClusterScope, awsMachine *infrav1.AWSMachine) (*scope.MachineScope, error) {
+	return scope.NewMachineScope(
+		scope.MachineScopeParams{
+			Client: testEnv,
+			Cluster: &clusterv1.Cluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test",
+				},
+				Status: clusterv1.ClusterStatus{
+					InfrastructureReady: true,
+				},
+			},
+			Machine: &clusterv1.Machine{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test",
+				},
+				Spec: clusterv1.MachineSpec{
+					Bootstrap: clusterv1.Bootstrap{
+						DataSecretName: pointer.StringPtr("bootstrap-data"),
+					},
+				},
+			},
+			InfraCluster: cs,
+			AWSMachine:   awsMachine,
+		},
+	)
+}
+
+func createAWSMachine(g *WithT, awsMachine *infrav1.AWSMachine) {
+	g.Expect(testEnv.Create(ctx, awsMachine)).To(Succeed())
+	g.Eventually(func() bool {
+		machine := &infrav1.AWSMachine{}
+		key := client.ObjectKey{
+			Name:      awsMachine.Name,
+			Namespace: awsMachine.Namespace,
+		}
+		err := testEnv.Get(ctx, key, machine)
```

Comment:
We can also write it as `return testEnv.Get(ctx, key, machine) == nil`

Reply:
Done